### PR TITLE
paywall modal ui enhancements according to issue description

### PR DESCRIPTION
**Pull request by CoDigo93**
that's a printscreen that shows how the new paywall modal looks after changing the cancel button color, and i did some size adjustments.

Captura de ecrã de 2021-09-03 16-07-36

Here is the next screen after clicking on the ''purchase button''
Captura de ecrã de 2021-09-03 16-14-20

Captura de ecrã de 2021-09-03 16-14-27

Clicking outside the modal container, closes the modal returning to the main page.
CPT2109031635-991x662

### CODE REVIEW

**Comment by mikhael28** [comment]
Hi @CoDigo93 thank you for taking up this ticket! Great work so far, the red on the cancel button is perfect. Can you make sure to pull the latest changes from `main`, and resolve the merge conflicts that are in `index.css` and `PaywallModal`? There have been a lot of changes in the last few days since you forked the repository :) for example, the default color for buttons is now green!

Then, can you please make sure that the initial video is fully extended, along the width of the paywall? The right side has white space, but the left side doesn't? 
Screen Shot 2021-09-03 at 5 02 53 PM

**Comment by mikhael28** [comment]
One thing to keep in mind for the future - someone already volunteered to do the ticket, and I assigned them to it. In the future, please do not start working on tickets that someone else has already agreed to do. I'm fine with it this time, but in the future please be conscious of who has been assigned to the ticket. If they submit a pull-request that is more complete/detailed/better written than yours before we finish with your code review, theirs will be merged :)

**Comment by CoDigo93** [comment]
> One thing to keep in mind for the future - someone already volunteered to do the ticket, and I assigned them to it. In the future, please do not start working on tickets that someone else has already agreed to do. I'm fine with it this time, but in the future please be conscious of who has been assigned to the ticket. If they submit a pull-request that is more complete/detailed/better written than yours before we finish with your code review, theirs will be merged :)

of course! Sorry for that, when i forked this repository this task was free, i just have some few hours in a day to take side projects like this . I'd have to refresh the page to see if there were new stuffs running here, sorry again! Next time i'll pay attention to that.

**Comment by mikhael28** [comment]
No worries @CoDigo93 - go ahead and make that whitespace fix, pull from the upstream repository to get the latest changes - reference this if you are unsure what I'm talking about: https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Then push those changes to your current branch, and they should show up on the pull-request.

Essentially, what I'm asking is for your downstream code, to have as much 'up to date code' as the upstream branch you are pushing code INTO. That way, the code that you are contributing will be clear, simple and obvious! And won't be out of date in any way.

**Comment by mikhael28** [comment]
Let me know if you have any questions about that process.

**Comment by CoDigo93** [comment]
The white space was fixed, what do you think about put some gap between the text and the video container?

Captura de ecrã de 2021-09-04 11-26-17

File: src/index.css [hunk 1]
```diff
@@ -434,10 +462,30 @@ input[type="file"] {
...
+  border-radius: 6px;
+  background-color: #f44336;
+  font-size: 18px;
+  
+}
+
 @media all and (min-width: 480px) {
   .Form {
     padding: 60px 0;
-    margin: 0 auto;
```
Comment (mikhael28) [inline src/index.css:489]: Please add this back in - you are breaking the form for login. It should be centered.
Screen Shot 2021-09-04 at 9 00 10 AM

File: src/learn/PaywallModal.js [hunk 2]
```diff
@@ -12,6 +13,25 @@ import Button from "react-bootstrap/lib/Button";
 
 function LoadingModal(props) {
   const [showPayment, setShowPayment] = useState(false);
+
+  const history = useHistory();
+  const modalRef = useRef();
+
+  useEffect(()=>{
+    let handleModalClose = (event) =>{
+      if(!modalRef.current?.contains(event.target)){
```
Comment (mikhael28) [inline src/learn/PaywallModal.js]: This functionality is meant to say 'if you click outside of the area, then close modal go to home' - however, it didn't work for me. Can you try to re-test it? Reference the `Modal` component in the `components` folder, for a work example.

File: src/index.css [hunk 3]
```diff
@@ -366,6 +366,7 @@ input[type="file"] {
 
 .btn {
   border: 0;
+  width:100%;
```
Comment (mikhael28) [inline src/index.css]: I don't like these types of edits - you are editing global css for a button, just to change a couple of buttons on your PR. Can you explain the reasoning for this? Have you gone through and checked every single button throughout the website, to check if it's still okay and not warped/ugly?

File: src/index.css [hunk 4]
```diff
@@ -413,8 +413,8 @@ input[type="file"] {
 
 .btn {
   border: 0;
-  width:100%;
-  background: rgb(37, 38, 39);
+  background: #339c51;
```
Comment (mikhael28) [inline src/index.css:416]: Ahh okay, I see you added that back in. Disregard that other comment.

File: src/learn/PaywallModal.js [hunk 5]
```diff
@@ -17,8 +17,10 @@ function LoadingModal(props) {
 
   useEffect(()=>{
     let handleModalClose = (event) =>{
-      if(!modalRef.current?.contains(event.target)){
-        history.push('/'); 
+      if(showPayment && !modalRef.current?.contains(event.target)){
```
Comment (mikhael28) [inline src/learn/PaywallModal.js:20]: Maybe this is why it stopped working? To clarify, it needs to redirect you back to home if regardless if you are on `showPayment` render or not. Just, if the paywall modal is up period - you should be able to leave.

File: src/learn/PaywallModal.js [hunk 6]
```diff
@@ -34,47 +36,41 @@ function LoadingModal(props) {
...
+            <Button className="btn-cancel" onClick={() => props.history.push("/")}>Cancel</Button>
+            <Button onClick={() => setShowPayment(true)}>Purchase</Button>
+          </div>
         </DialogContent>
       ) : (
         <DialogContent ref={modalRef} style={{ textAlign: "start" }}>
           <div className='ConfirmPurchaseForm'>
             <Order {...props} />
-            <Button onClick={() => setShowPayment(false)}>Back</Button>
+            <Button className="btn-cancel" onClick={() => setShowPayment(false)}>Back</Button>
```
Comment (mikhael28) [inline src/learn/PaywallModal.js:73]: Yet we have a cancel button now! Thank you :D

**Review by mikhael28 — CHANGES_REQUESTED** [review]
Hey there, great work so far - I'm requesting some changes, as some of the CSS you added broke the Login page, and also the Stripe credit card for is totally broken now. here is what it looks like: it's a little square. Would you fix that please?
Screen Shot 2021-09-04 at 9 00 02 AM

File: src/index.css (hunk 1 above)
Comment (CoDigo93) [reply src/index.css:489]: That's how the login page should be?

Captura de ecrã de 2021-09-04 14-15-12
 be?

File: src/learn/PaywallModal.js (hunk 2 above)
Comment (CoDigo93) [reply src/learn/PaywallModal.js:20]: It is working now!That 'showPayment' verification was breaking the feature.

CPT2109041429-1152x664

File: src/index.css (hunk 1 above)
Comment (CoDigo93) [reply src/index.css:489]: that image of credit card field refers to how it should be ? if dont, can you show me how it is supposed to be?

Comment (mikhael28) [reply src/index.css:489]: Yes, that's how the login form should look. Regarding the credit card - checking now, it looks like the Order component is broken for some reason. Will need to fix that - just put the credit card form back the way it was previously, and I will handle that myself.

File: src/learn/PaywallModal.js (hunk 2 above)
Comment (mikhael28) [reply src/learn/PaywallModal.js:20]: Is your code breaking when you click 'Purchase'? Mine is... on the production site.

Comment (CoDigo93) [reply src/learn/PaywallModal.js:20]: no, its working

CPT2109041825-832x629

File: src/index.css (hunk 1 above)
Comment (CoDigo93) [reply src/index.css:489]: Now that is how the form looks, now i can see how the shape of the credit card field must be, i had to create a button outside the container, that may have broken the layout 

Captura de ecrã de 2021-09-04 18-42-12

File: src/learn/PaywallModal.js (hunk 2 above)
Comment (mikhael28) [reply src/learn/PaywallModal.js:20]: That was a bug that I just fixed haha it was breaking in production for some reason.

Comment (mikhael28) [reply src/learn/PaywallModal.js:20]: Variable propagation wasn't quite working, changed the stripeKey to a prop from paywall modal and it's all good

**Comment by mikhael28** [comment]
@CoDigo93 I will take another look in 15 minutes, but I think we are good to go (pending some weird technical issue I haven't foreseen) - home stretch! Will have an update shortly

**Comment by mikhael28** [comment]
Alright @CoDigo93 your code has been merged!!! Great work, way to stick through it was all the feedback :D

**Comment by mikhael28** [comment]
Can you introduce yourself in #17 ?

**Pull request merged**